src/phases/faint-phase.ts

```diff
@@ -55,8 +55,10 @@ export class FaintPhase extends PokemonPhase {
     // Track total times pokemon have been KO'd for supreme overlord/last respects
     if (pokemon.isPlayer()) {
       this.scene.currentBattle.playerFaints += 1;
+      this.scene.currentBattle.playerFaintsHistory.push({ pokemon, turn: this.scene.currentBattle.turn });
     } else {
       this.scene.currentBattle.enemyFaints += 1;
+      this.scene.currentBattle.enemyFaintsHistory.push({ pokemon, turn: this.scene.currentBattle.turn });
```

Comment:
nit: might be good for clarity here
```suggestion
      this.scene.currentBattle.playerFaintsHistory.push({ pokemon: pokemon, turn: this.scene.currentBattle.turn });
    } else {
      this.scene.currentBattle.enemyFaints += 1;
      this.scene.currentBattle.enemyFaintsHistory.push({ pokemon: pokemon, turn: this.scene.currentBattle.turn });
```